Tests/TuistCoreTests/Graph/GraphTraverserTests.swift

```diff
@@ -1617,7 +1617,6 @@ final class GraphTraverserTests: TuistUnitTestCase {
         // Then
         XCTAssertEqual(got, [
             GraphDependencyReference(dependencyPrecompiledStaticBinaryA),
-            GraphDependencyReference(dependencyPrecompiledStaticBinaryB),
```

Comment:
I think that `StaticA` should contain the symbols of `StaticB` 👍 
I'll extend the fixture to confirm 🚀

Reply:
Just thinking again about it, also in case of precompiled targets, if we don't link `StaticB` to `App`, then `App` will not be able to use `StaticB` symbols, which might be semantically correct as it's not defining a dependency on it, but it's not what the apple ecosystem usually do.
What do you think of it, do you think it's better to leave everything as it is? 🤔 
Yes IF not precompiled we should include them, as StaticA only contains App can import

Reply:
It's best to define explicit dependencies rather than rely on implicit behaviour.

The determining factor would be if `StaticA.xcframework` has all the symbols it needs from `StaticB` or if the `App` must link them - from what I've seen in the `FirebaseAnalytics` example in the sample project of https://github.com/tuist/tuist/pull/4170 indicates to me that isn't the case - but could relate to how the xcframework is built 🤔

Reply:
@fortmarek @kwridan I have added the scenarios to the fixture, it seems to work fine 🚀
Let me know what you think about it 🙏